src/main/java/ch/njol/skript/conditions/CondItemLifetime.java

```diff
@@ -0,0 +1,55 @@
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Item;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Item Has Unlimited Lifetime")
+@Description("Checks if the dropped item has unlimited lifetime enabled or disabled.")
+@Examples({
+	"if all dropped items have unlimited lifetime disabled:",
+		"\tenable unlimited lifetime for all dropped items"
+})
+@Since("INSERT VERSION")
+public class CondItemLifetime extends Condition {
+
+	static {
+		Skript.registerCondition(CondItemLifetime.class, ConditionType.PROPERTY,
+			"[the] %itementities% (has|have) (unlimited|infinite) life(time|span) [enabled]",
+			"(unlimited|infinite) life(time|span) (is|are) enabled for %itementities%",
+			"[the] %itementities% (has|have) (unlimited|infinite) life(time|span) disabled",
+			"(unlimited|infinite) life(time|span) (is|are) disabled for %itementities%",
+			"[the] %itementities% (don't|do not|doesn't|does not) have (unlimited|infinite) life(time|span)");
```

Comment:
i really am not a fan of the enabled/disabled being in the syntax, it doesn't make much sense to me.
if you stick wit just has/has not this can be a PropertyCondition, but honestly it's pretty awkward even then with the effect.
I'd much rather see something like `if x will/won't naturally despawn` and `prevent x from naturally despawning` or similar, or something like `will age` like how paper's methods are named.